agenta-cli/agenta/sdk/config_manager.py

```diff
@@ -0,0 +1,205 @@
+
+        Note:
+            The method prioritizes the inputs in the following way:
+            1. 'config' (i.e. when called explicitly from the playground)
+            2. 'environment'
+            3. 'variant'
+            Only one of these should be provided.
+        """
+        context = route_context.get()
+        if ("config" in context and context["config"]) and (
+            ("environment" in context and context["environment"])
+            or ("variant" in context and context["variant"])
+        ):
+            raise ValueError(
+                "Either config, environment or variant must be provided. Not both."
+            )
+        if "config" in context and context["config"]:
+            return schema(**context["config"])
+        elif "environment" in context and context["environment"]:
+            return ConfigManager.get_from_backend(
+                schema, environment=context["environment"]
+            )
+        elif "variant" in context and context["variant"]:
+            return ConfigManager.get_from_backend(schema, variant=context["variant"])
+        else:
+            raise ValueError("Either config, environment or variant must be provided")
+
+    @staticmethod
+    def get_from_backend(
+        schema: Type[T],
+        environment: Optional[str] = None,
+        version: Optional[str] = None,
+        variant: Optional[str] = None,
+    ) -> T:
+        """
+        Pulls the parameters for the app variant from the server and returns a config object.
+
+        This method retrieves the configuration from the backend server based on the provided
+        environment or variant. It then validates and returns the configuration as an instance
+        of the specified schema.
+
+        Args:
+            schema (Type[T]): A Pydantic model class that defines the structure of the configuration.
+            environment (Optional[str]): The environment name to fetch the configuration for.
+                                         Must be one of "development", "production", or "staging".
+            version (Optional[str]): Currently not implemented. Will raise NotImplementedError if provided.
+            variant (Optional[str]): The variant name to fetch the configuration for.
+
+        Returns:
+            T: An instance of the specified schema populated with the configuration data.
+
+        Raises:
+            ValueError: If neither environment nor variant is provided.
+            NotImplementedError: If a specific version is requested (not yet implemented).
+            ValidationError: If the retrieved configuration data doesn't match the schema.
+            Exception: For any other errors during the process (e.g., API communication issues).
+
+        Note:
+            Either environment or variant must be provided, but not both.
+        """
+        if not ConfigManager.client:
+            try:
+                ConfigManager.client = AgentaApi(
```

Comment:
This would include the project-id at some point, right ?
Do we want to add it now ?

Reply:
It would be hard to do it now, since we neither know how we will identify the project (id? slug?) + the client here is auto-generated with fern

Reply:
Yes -> at some point.